module/os/linux/zfs/vdev_disk.c

```diff
@@ -209,50 +229,47 @@ static void
 vdev_disk_kobj_evt_post(vdev_t *v)
 {
 	vdev_disk_t *vd = v->vdev_tsd;
-	if (vd && vd->vd_bdev) {
-		spl_signal_kobj_evt(vd->vd_bdev);
+	if (vd && vd->vd_bdh) {
+		spl_signal_kobj_evt(BDH_BDEV(vd->vd_bdh));
 	} else {
 		vdev_dbgmsg(v, "vdev_disk_t is NULL for VDEV:%s\n",
 		    v->vdev_path);
 	}
 }
 
-#if !defined(HAVE_BLKDEV_GET_BY_PATH_4ARG)
-/*
- * Define a dummy struct blk_holder_ops for kernel versions
- * prior to 6.5.
- */
-struct blk_holder_ops {};
-#endif
-
-static struct block_device *
-vdev_blkdev_get_by_path(const char *path, spa_mode_t mode, void *holder,
-    const struct blk_holder_ops *hops)
+static zfs_bdev_handle_t *
+vdev_blkdev_get_by_path(const char *path, spa_mode_t mode, void *holder)
 {
-#ifdef HAVE_BLKDEV_GET_BY_PATH_4ARG
+#if defined(HAVE_BDEV_OPEN_BY_PATH)
+	return (bdev_open_by_path(path,
+	    vdev_bdev_mode(mode, B_TRUE), holder, NULL));
+#elif defined(HAVE_BLKDEV_GET_BY_PATH_4ARG)
 	return (blkdev_get_by_path(path,
-	    vdev_bdev_mode(mode, B_TRUE), holder, hops));
+	    vdev_bdev_mode(mode, B_TRUE), holder, NULL));
 #else
 	return (blkdev_get_by_path(path,
 	    vdev_bdev_mode(mode, B_TRUE), holder));
 #endif
 }
 
 static void
-vdev_blkdev_put(struct block_device *bdev, spa_mode_t mode, void *holder)
+vdev_blkdev_put(zfs_bdev_handle_t *bdh, spa_mode_t mode, void *holder)
 {
-#ifdef HAVE_BLKDEV_PUT_HOLDER
-	return (blkdev_put(bdev, holder));
+#if defined(HAVE_BDEV_OPEN_BY_PATH)
```

Comment:
Since both interfaces were added at the same time this works, but we should probably be checking `HAVE_BDEV_RELEASE` here.

Reply:
Oh yep, it was always supposed to be that, good spot. Fixed.